velox/common/caching/tests/AsyncDataCacheTest.cpp

```diff
@@ -572,6 +584,16 @@ TEST_F(AsyncDataCacheTest, outOfCapacity) {
   EXPECT_EQ(4092, cache_->numAllocated());
 }
 
+namespace {
+// Cuts off the last 1/10th of file at 'path'.
+void corruptFile(const std::string& path) {
+  auto fd = open(path.c_str(), O_WRONLY);
+  auto size = lseek(fd, 0, SEEK_END);
+  auto rc = ftruncate(fd, size / 10 * 9);
+  EXPECT_EQ(0, rc);
+}
+} // namespace
+
 TEST_F(AsyncDataCacheTest, ssd) {
```

Comment:
to make test failures more actionable, it would be nice if we could break this large test case into multiple, smaller ones. For instance, we could have one "simple" test case that writes a few files and checks if we can recover them, a different one doing it in parallel, and another one that tests we can recover files with a corrupted tail. The idea being that when something is broken, the test case name could already give you a pretty good idea about what got broken.